src/cpp/session/modules/SessionRMarkdown.R

```diff
@@ -111,21 +118,29 @@
 # whether it's up to date (e.g. for input.Rmd producing input.html, see whether
 # input.html exists and has been written since input.Rmd)
 .rs.addFunction("getRmdOutputInfo", function(target) {
-  # compute the name of the output file
-  lines <- readLines(target, warn = FALSE)
-  outputFormat <- rmarkdown:::output_format_from_yaml_front_matter(lines)
-  outputFormat <- rmarkdown:::create_output_format(
-                                    outputFormat$name, outputFormat$options)
-  outputFile <- rmarkdown:::pandoc_output_file(target, outputFormat$pandoc)
-  outputPath <- file.path(dirname(target), outputFile) 
-
-  # ensure output file exists
-  current <- file.exists(outputPath) && 
-             file.info(outputPath)$mtime >= file.info(target)$mtime
   
-  return(list(
-    output_file = .rs.scalar(outputPath),
-    is_current  = .rs.scalar(current)))
+   # read the contents of the file
+   lines <- readLines(target, warn = FALSE)
+   
+   # mark the encoding if it's available
+   properties <- .rs.getSourceDocumentProperties(target)
+   if (identical(properties$encoding, "UTF-8"))
+      Encoding(lines) <- "UTF-8"
+   
+   # compute the name of the output file
+   outputFormat <- rmarkdown:::output_format_from_yaml_front_matter(lines)
```

Comment:
The diffs here are just indentation changes -- for some reason this part of the document was indented with a two-space indent.